Framework/Source/Utils/Gui.cpp

```diff
@@ -365,6 +395,62 @@ namespace Falcor
         return b;
     }
 
+    bool Gui::addInt2Var(const char label[], glm::ivec2& var, int minVal, int maxVal, bool sameLine)
+    {
+        if (sameLine) ImGui::SameLine();
+        bool b = ImGui::InputInt2(label, static_cast<int*>(glm::value_ptr(var)), 0);
+        var = clamp(var, minVal, maxVal);
+        return b;
+    }
+
+    bool Gui::addInt3Var(const char label[], glm::ivec3& var, int minVal, int maxVal, bool sameLine)
+    {
+        if (sameLine) ImGui::SameLine();
+        bool b = ImGui::InputInt3(label, static_cast<int*>(glm::value_ptr(var)), 0);
+        var = clamp(var, minVal, maxVal);
+        return b;
+    }
+
+    bool Gui::addInt4Var(const char label[], glm::ivec4& var, int minVal, int maxVal, bool sameLine)
+    {
+        if (sameLine) ImGui::SameLine();
+        bool b = ImGui::InputInt4(label, static_cast<int*>(glm::value_ptr(var)), 0);
+        var = clamp(var, minVal, maxVal);
+        return b;
+    }
+
+#define concatStrings_(a, b) a##b
+#define concatStrings(a, b) concatStrings_(a, b)
+#define add_matrix_function(funcName, matrixSize, baseFunc) \
+    bool concatStrings(Gui::, funcName) (const char label[], concatStrings(glm::mat, matrixSize) & var, float minVal, float maxVal, bool sameLine) \
+    { \
+        std::string labelString(label); \
+        labelString.append("[0]"); \
+        bool b = false; \
+        \
+        for (int i = 0; i < var.length(); ++i) \
```

Comment:
`.length()` returns a `size_t`, which would be a signed/unsigned mismatch when comparing with `i`. MSVC isn't picky about this, but GCC will be.